### PR TITLE
Add/update examples with new functionality

### PR DESCRIPTION
**Pull request by dmkozh**
### What

- Change deployer example to deploy contracts on behalf of any address.
- Add example for deep sub-contract authorization

### Why

Demonstrates the new contract deployment capabilities

### Known limitations

N/A

<a href="https://gitpod.io/#https://github.com/stellar/soroban-examples/pull/261"></a>

### CODE REVIEW

File: deep_contract_auth/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[package]
```
Comment (sisuresh) [inline deep_contract_auth/Cargo.toml:1]: The Makefile for this new example is missing.

Comment (dmkozh) [reply deep_contract_auth/Cargo.toml:1]: It's kind of trivial as this example is not intended to be built as Wasm (that would require 3 separate contracts, which is way beyond the point of what this demonstrates). But added for consistency.

**Review by sisuresh — APPROVED** [review]

File: deep_contract_auth/Cargo.toml (hunk 1 above)
Comment (sisuresh) [reply deep_contract_auth/Cargo.toml:1]: Is that why the Makefile uses `cargo build` instead of `soroban contract build`?

Comment (dmkozh) [reply deep_contract_auth/Cargo.toml:1]: Yeah, I don't want to build wasm for this one (I'm not even sure what would happen).

**Pull request merged**